src/operators/mod.rs

```diff
@@ -65,6 +65,39 @@ pub(crate) fn filter(predicate: &TdPyCallable, item: &TdPyAny) -> bool {
     Python::with_gil(|py| with_traceback!(py, predicate.call1(py, (item,))?.extract(py)))
 }
 
+pub(crate) fn fold_epoch(
+    builder: &TdPyCallable,
+    folder: &TdPyCallable,
+    acc: &mut Option<TdPyAny>,
+    _key: &StateKey,
+    value: TdPyAny,
+) {
+    debug!(
+        "{}, builder:{:?}, folder:{:?}, key:{:?}, value:{:?}, acc:{:?}",
+        log_func!(),
+        builder,
+        folder,
+        _key,
+        value,
+        acc
+    );
+    Python::with_gil(|py| {
+        let updated_acc = match acc {
+            Some(acc) => with_traceback!(py, folder.call1(py, (acc.clone_ref(py), value))).into(),
+            None => with_traceback!(py, builder.call1(py, (value, ))).into(),
```

Comment:
Ok.
Just calling it like you propose makes the borrow checker not happy (cannot move out of `*acc` which is behind a mutable reference), and trying to borrow the option's value (like `acc.as_ref()....`) makes the `.into()` call not work because the trait `&TdPyAny: From<pyo3::Py<pyo3::PyAny>>` is not satisfied (only without the reference).

If I `.take()` the whole `Option` out it works though, and I can also avoid the `clone_ref(py)` when calling the folder:

```rust
let initial_acc = acc.take().unwrap_or_else(|| {
    with_traceback!(py, builder.call1(py, (value.clone_ref(py),))).into()
});
let updated_acc = with_traceback!(py, folder.call1(py, (initial_acc, value))).into();
*acc = Some(updated_acc);

```
Since I always set the new acc value, this should be fine.

Reply:
Yep, that looks great. I need the borrow checker to check everything I write.